### DISCUSSION TITLE
Config for "automerge patch always, minor if approved by review"?

### DISCUSSION THREAD

**Discussion by leaumar** [renovatebot/renovate#9120] [Request Help]
I run the renovate docker image on git*lab* CI nightly. Current config:
```
{
    "extends": ["config:base", ":automergeBranch", ":automergePatch", ":automergeTypes", ":pinVersions"]
}
```
My repositories have no mandatory PR approval since they're private projects, so this config nicely does the following:

- silently merges patches
- makes a PR for minors so I can read about new features or changes (usually)
- makes a PR for any failing updates

However, it does sometimes create a PR for a minor that immediately ends up having to be rebased already (and sometimes can't be rebased) because a patch is merged during the same run that created the PR. I like to get these PRs out of the way ASAP, but I often have to either rebase them all and then wait around a while (to either merge them all, or let gitlab automerge them and delete the mail spam that follows), or wait around for an extra manual Renovate run before merging them all.

So I'd like to add the following behavior:

- make a PR for minors but automerge them (after a rebase or do-over during the next run if necessary) if the PR is approved (while PR approval remains optional in the repo)

![image](https://user-images.githubusercontent.com/3950300/111042173-568e4280-843c-11eb-823c-5d96a72145ef.png)
(figure: I wanna hit that Approve button and let renovate automerge overnight, instead of rebasing or waiting for an overnight do-over)

I had a look through the doc but I can't seem to piece together the config to achieve this. I'd envision something like
```
"minor": {
  "automerge": true,
  "automergeType": "pr",
  "requiredStatusChecks": {
    "minimumPrApprovals": 1
  }
}
```
but [`requiredStatusChecks`](https://docs.renovatebot.com/configuration-options/#requiredstatuschecks) has no such nuance yet AFAIK.

**Comment by rarkins** [comment]
You're right, there's no such concept yet that automerging requires an approval. Does GitLab persist approvals even after new commits are made? In GitHub approvals get dismissed as soon as Renovate rebases its own branch.

**Reply by leaumar** [reply]
Hm, good point, approval probably does expire with changes and renovate would have to remember the approval 2 runs later (1 to rebase, 1 to merge after CI). It would need to work equally on github anyway so that idea's out the window.

`automergeType=pr-comment` is when renovate tells another bot to merge, so perhaps a new feature could be the inverse, to substitute the approval? E.g. `automergeType=on-demand` with a `mergeCommand="ship it!"` (or, more conveniently, "k" 🙂 ), making renovate automerge the PR only if a comment equal to that text is posted on it.

**Reply by leaumar** [reply]
Hang on though, there's a simpler solution. Renovate already puts that "tick me to rebase" checkbox in the PR body, it could simply have a "tick me to approve this PR and allow automerge after an update if this PR is not mergeable right now" checkbox? 🙂

**Reply by rarkins** [reply]
@leaumar sounds like this: https://github.com/renovatebot/renovate/issues/3172 ?

**Reply by leaumar** [reply]
Yes, and my first idea sounds like #1355 as well. I'll follow those convos.

Thanks @rarkins
